include/zephyr/bluetooth/audio/pbp.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_BLUETOOTH_AUDIO_PBP_
+#define ZEPHYR_INCLUDE_BLUETOOTH_AUDIO_PBP_
+
+#include <zephyr/sys/util.h>
+
+#include <zephyr/bluetooth/audio/audio.h>
+
+/** Public Broadcast Announcement features */
+enum bt_pbp_announcement_feature {
+	/** Broadcast Streams encryption status */
+	BT_PBP_ANNOUNCEMENT_FEATURE_ENCRYPTION = BIT(0),
+	/** Standard Quality Public Broadcast Audio configuration */
+	BT_PBP_ANNOUNCEMENT_FEATURE_STANDARD_QUALITY = BIT(1),
+	/** High Quality Public Broadcast Audio configuration */
+	BT_PBP_ANNOUNCEMENT_FEATURE_HIGH_QUALITY = BIT(2),
+};
+
+/**
+ * @brief Sets the received Public Broadcast Announcement features.
+ *
+ * @param features	Public Broadcast Announcement features
+ *
+ */
+void bt_pbp_init_source(enum bt_pbp_announcement_feature features);
+
+/**
+ * @brief Creates a Public Broadcast Announcement based on the information received
+ * in bt_pbp_init_source().
+ *
+ * @param meta		Metadata to be included in the advertising data
+ * @param meta_len	Size of the metadata fields to be included in the advertising data
+ * @param pba_data_buf	Pointer to store the PBA advertising data. Buffer size needs to be
+ *			meta_len + 3 (service UUID + PBA feature + metadata).
+ *
+ * @return 0 on success or an appropriate error code.
+ */
+int bt_pbp_create_public_broadcast_announcement(const uint8_t meta[], size_t meta_len,
+						struct net_buf_simple *pba_data_buf);
+
+/**
+ * @brief Parses the received advertising data corresponding to a Public Broadcast
+ * Announcement. Returns the advertised Public BroadcastAnnouncement features.
+ * and metadata.
+ *
+ * @param data			Advertising data to be checked
+ * @param source_features	Sink stream configuration preferences
+ * @param meta			Pointer to copy the metadata present in the advertising data
+ *
+ * @return parsed metadata length on success or an appropriate error code
+ */
+int bt_pbp_parse_public_broadcast_announcement(struct bt_data *data,
+					       enum bt_pbp_announcement_feature *source_features,
+					       uint8_t *meta);
```

Comment:
Ditto: We may want to use `pba` instead of `public_broadcast_announcement`

Reply:
PBA is the official acronym for Public Broadcast Assistant, so I avoided it to not cause confusion

Reply:
Maybe just a shorthand for `announcement` would suffice

Reply:
Alternatively maybe we can just use `announcement` without `public_broadcast` as that is implied by the prefix. 

So we could do `bt_pbp_get_announcement` and `bt_pbp_parse_announcement`